frontend/src/composables/use-component-name.ts

```diff
@@ -0,0 +1,10 @@
+import { getCurrentInstance } from "vue"
+
+/**
+ * Retrieve the name of the component in which
+ * the composable is being used.
+ */
+export const useComponentName = () => {
+  const vm = getCurrentInstance()
+  return vm?.proxy.$options.name
```

Comment:
I don't think `name` is a required property so having a fallback in case it's `undefined` might be a good idea. A lint check to ensure all our components have names would be a better idea but that might be more work.

Reply:
Another thought: should `useComponentName` raise an error if it's used in a context where no component name exists? That would keep the return value stable `string` not `undefined | string`, but would be much stricter than this implementation.

Falling back to "Unknown" without it being intentional, especially if this is used to disambiguate events, doesn't seem like a good long-term practice.

Reply:
I would like to have a stricter implementation, but when we test the `use-search-type` in unit tests, there is no component, so this will throw errors. Maybe we could create a new issue to see if we have many "Unknown" components in analytics reports, and if we do - fix it?